lib/project_types/script/layers/infrastructure/languages/assemblyscript_project_creator.rb

```diff
@@ -11,23 +11,73 @@ class AssemblyScriptProjectCreator
           property! :script_name, accepts: String
           property! :path_to_project, accepts: String
 
-          BOOTSTRAP = "npx --no-install shopify-scripts-toolchain-as bootstrap --from %{extension_point} --dest %{base}"
           BUILD = "shopify-scripts-toolchain-as build --src src/shopify_main.ts " \
           "--binary build/script.wasm --metadata build/metadata.json"
-          MIN_NODE_VERSION = "14.5.0"
+          MIN_NODE_VERSION = "14.5.0" # kept because task_runner uses this
           ASC_ARGS = "-- --lib node_modules --optimize --use Date="
 
+          EP_REPO = "https://github.com/Shopify/extension-points.git"
+          ORIGIN_BRANCH = 'master'
+
           def setup_dependencies
+            # sparse checkout actions
+            git_init
+            setup_remote
+            setup_sparse_checkout
+            pull
+            clean
+
+            # TODO: files that should be in the repo, but arent yet
             write_npmrc
             write_package_json
           end
 
           def bootstrap
-            command_runner.call(bootstap_command)
           end
 
           private
 
+          def git_init
+            out, status = ctx.capture2e("git init")
+            raise Domain::Errors::ServiceFailureError, out unless status.success?
+          end
+
+          def setup_remote
+            out, status = ctx.capture2e("git remote add -f origin #{EP_REPO}")
+            raise Domain::Errors::ServiceFailureError, out unless status.success?
+          end
+  
+          def setup_sparse_checkout
+            out, status = ctx.capture2e("git config core.sparsecheckout true")
+            raise Domain::Errors::ServiceFailureError, out unless status.success?
+
+            out, status = ctx.capture2e("git sparse-checkout set #{sparse_checkout_set_path}")
+            raise Domain::Errors::ServiceFailureError, out unless status.success?
+          end
+
+          def sparse_checkout_set_path
+            type = extension_point.dasherize_type
+            domain = extension_point.domain
+
+            if domain.nil?
+              "packages/default/extension-point-as-#{type}/assembly/sample"
+            else
+              "packages/#{domain}/samples/#{type}"
+            end
+          end
+  
+          def pull
+            out, status = ctx.capture2e("git pull origin #{ORIGIN_BRANCH}")
+            raise Domain::Errors::ServiceFailureError, out unless status.success?
+          end
+  
+          def clean
+            source = File.join(path_to_project, sparse_checkout_set_path)
+            FileUtils.copy_entry(source, path_to_project)
+            ctx.rm_rf("packages")
+            ctx.rm_rf(".git")
```

Comment:
Are these the files that sparse check produces?  Is it important to delete them?

Reply:
It's not a result of sparse checkout per-se. Since sparse checkout effectively clones a remote directory, it will clone everything within the remote repo, we delete this in order to avoid "polluting" the script project with the git information from the source repository. We chatted offline a bit more about this and another way to achieve the same result is to apply filtering when performing the sparse checkout call. I think that such an approach is cleaner.

Reply:
@saulecabrera Just so I am clear, you mean the "filter param" is the cleaner approach?

Reply:
Yeah, that's right.

Reply:
I was a bit confused by this comment chain. This PR uses sparse checkout with the right "filter" (the set <path> command). Was there another method of filtering that I missed? Or am I ok to resolve this comment chain?